CHANGELOG.md

```diff
@@ -947,6 +947,7 @@
   projects next to each other][7634]
 - [Support runtime checks of intersection types][7769]
 - [Merge `Small_Integer` and `Big_Integer` types][7636]
+- [New `project/status` route for reporting LS state][7801]
```

Comment:
I don't see change in the language server protocol specification. When at it, can you also start versioning the protocol? At least include semantic version to the header of the document and mention that `project/status` message is available _since_ that version.

Here is an [apichanges inspiration](https://bits.netbeans.org/16/javadoc/org-openide-util-lookup/apichanges.html).

Reply:
I added documentation for the protocol but not the versioning. I don't think the latter has been fully agreed and feels wrong to suddenly plug it into this PR without adding logic that would negotiate protocol between LS and IDE. I think it's a good idea in general just not in this PR which is already not trivial to review.

Besides, the current approach seems to be to "always use latest" and its not breaking the protocol, just adding a route.

Reply:
We had a discussion about versioning #7639, and I think, instead of versioning the protocol, it would be reasonable to make IDE use the latest language server API (because they are supposed to work together), and just track the oldest engine version that the current language server supports.

Reply:
Compatibility is about **attitude**. Once we start our effort to be compatible we'll be (mostly) compatible. Until we start any effort (e.g. at least versioning of the protocol documentation!), we'll be **incompatible**.

> I don't think the latter has been fully agreed on

Let me know what kind of approval you need for starting the semantic versioning of the language server specification.

Reply:
> Let me know what kind of approval you need for starting the semantic versioning of the language server specification.

I don't want to add versioning in this PR as its not directly related to what this is trying to fix. The whole versioning story is part of the #7639 discussion and #7717 ticket. 

But I do need your approval, @JaroslavTulach, since I can't merge this fix otherwise.